numpy/core/src/multiarray/compiled_base.c

```diff
@@ -605,6 +664,12 @@ arr_interp(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwdict)
                 const npy_double slope = (slopes != NULL) ? slopes[j] :
                                          (dy[j+1] - dy[j]) / (dx[j+1] - dx[j]);
                 dres[i] = slope*(x_val - dx[j]) + dy[j];
+                /* Sometimes nan should be some other value */
+                if (npy_isnan(dres[i])) {
+                    dres[i] = _non_finite_interp(
```

Comment:
Doesn't:
```
if (npy_isnan(dres[i]) && (slope == 0)) {
    /* `x_val - dx[j] may not be finite, so interpolate from dx[j+1] instead */
    dres[i] = slope * (dx[j+1] - x_val) + dy[j+1];
}
```
do the trick?  Assuming the compiler does not optimize out 0 * (not finite), otherwise should do an `npy_isfinite(dx[j+1] - x_val)` check instead. Will probably just try myself in a bit though.

Started wondering about cases where `x_val - dx[j]` is infinite but `x_val and dx[j]` are not. So I think the function above should likely use the differences for its logic if something like my logic is not sufficient.

An `NPY_UNLIKELY` might make sense, though likely also no difference.

Reply:
Not sufficient - most of the important cases to cover are when `slope` is `nan` or `inf`. Thanks for the idea though - see #12989 for a simpler approach that comes with a little more overhead.